### PR TITLE
Cloud metadata

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Add `cloud` metadata to Elastic APM event schema. See https://github.com/elastic/apm/issues/256 for rationale.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] My code follows the style guidelines of this project (run `make check-full` for static code checks and linting)
- [x] I have rebased my changes on top of the latest master branch
~- [ ] I have made corresponding changes to the documentation~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally with my changes
- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes

1. Implement agent-side support for detecting and sending cloud metadata (https://github.com/elastic/apm/issues/256)
2. Have the agent send some transactions to the server
3. Check that the new fields are indexed

## Related issues

Closes https://github.com/elastic/apm-server/issues/3660

### CODE REVIEW

**Comment by axw** [comment]
jenkins run the tests please

File: _meta/fields.common.yml [hunk 1]
```diff
@@ -685,3 +685,80 @@
...
+          level: extended
+          type: keyword
+          ignore_above: 1024
+          description: Cloud account ID
+        - name: name
+          level: extended
+          type: keyword
+          ignore_above: 1024
+          description: Cloud account name
+      - name: availability_zone
```
Comment (graphaelli) [inline _meta/fields.common.yml:709]: I was expecting these to need `overwrite: true` - that doesn't seem to be enforced any more - this is worrying.

Comment (axw) [reply _meta/fields.common.yml:709]: So apparently it fails when you run `apm-server export index-pattern`. I'll fix this instance and create an issue to make sure this gets caught in CI.

Comment (axw) [reply _meta/fields.common.yml:711]: I've updated this file, and opened #3740 to fix the systemic problem.

**Pull request merged**